src/forms/renderers/Overrides/material/controls/MaterialDateControl.tsx

```diff
@@ -91,7 +85,11 @@ export const Custom_MaterialDateControl = (props: ControlProps) => {
                 }}
                 direction="row"
             >
-                <MaterialInputControl input={MuiInputText} {...props} />
+                <CustomMaterialInputControl
+                    inputEvents={events}
+                    input={MuiInputText}
+                    {...props}
+                />
```

Comment:
Using a custom input control so that we can control the actions happening on the input